I'm getting some issues trying to test the crate:

```
error[E0433]: failed to resolve: use of undeclared crate or module `diagnostic`
  --> /Users/owen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-core-0.4.0/src/extract/mod.rs:48:5
   |
48 |     diagnostic::on_unimplemented(
   |     ^^^^^^^^^^ use of undeclared crate or module `diagnostic`

error[E0433]: failed to resolve: use of undeclared crate or module `diagnostic`
  --> /Users/owen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-core-0.4.0/src/extract/mod.rs:75:5
   |
75 |     diagnostic::on_unimplemented(
   |     ^^^^^^^^^^ use of undeclared crate or module `diagnostic`

error[E0635]: unknown feature `diagnostic_namespace`
 --> /Users/owen/.cargo/registry/src/index.crates.io-6f17d22bba15001f/axum-core-0.4.0/src/lib.rs:1:45
  |
1 | #![cfg_attr(nightly_error_messages, feature(diagnostic_namespace))]
  |                                             ^^^^^^^^^^^^^^^^^^^^
```

Is axum core mismatched in version compared to axum proper?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,0 +1,24 @@
+name: Cargo Build & Test
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Rust project - latest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - stable
+          - beta
+          - nightly
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
+      - run: cargo build --verbose
+      - run: cargo test --verbose

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -14,7 +14,6 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 async-trait = "0.1.73"
-axum = "0.7"
 axum-core = "0.4"
 base64 = "0.21"
 http = "1"

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ Our version follows axum since 0.7. You can also enable just basic/bearer auth v
 axum-auth = { version = "0.7", default-features = false, features = ["auth-basic"] }
 ```
 
-If you're still using axum 0.5, use version 0.3. If you're still using axum 0.6, use version 0.5.
+If you're still using axum 0.5, use version 0.3. If you're still using axum 0.6, use version 0.4.
 
 ## Security
 

